### PR TITLE
fix: custom block context menus

### PR DESCRIPTION
**Pull request by BeksOmega**
## The basics

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

#5975 

### Proposed Changes

Moves the context menu property definition to the block instead of block svg.

#### Behavior Before Change

No custom context menus =(

#### Behavior After Change

Custom context menus!

### Reason for Changes

Any custom context menu function that was getting mixed into the block was getting overwritten by the block svg constructor. This fixes that.

### Test Coverage

Manually tested that the procedure context menus appear.

Tested on:
 * Desktop Chrome

### Documentation

N/A

### Additional Information

Thanks for the catch Christopher!

### CODE REVIEW

**Comment by BeksOmega** [comment]
Per offline discussion, I've moved the properties back into BlockSvg (since they are rendered-specific) and changed the assignment to be identity (eg `this.decompose = this.decompose`) to avoid overwriting mixed-in properties.

**Review by cpcallen — COMMENTED** [review]
As noted in our conversation:
* It's best to actually initialise any properties defined in a class constructor, to keep the shapes of the resulting objects consistent, but where a property is being redefined (to change its type) it is fine to just mention it without initialisation.
* It would be better not to move properties used only by `BlockSvg` to `Block`.
* Where a property in the subclass constructor is initialised to `undefined`, an initialisation like `this.prop = this.prop` will ensure it exists and is set to `undefined` if it was not set by the mixin.
* Changing the type of a superclass property in a subclass and user-defined mixins are both… problematic, as far as type-checking goes—but that is a problem for another ~~day~~ quarter.

File: tests/deps.js [hunk 1]
```diff
@@ -143,7 +143,7 @@ goog.addDependency('../../core/registry.js', ['Blockly.registry'], [], {'lang':
...
 goog.addDependency('../../core/renderers/common/constants.js', ['Blockly.blockRendering.ConstantProvider'], ['Blockly.ConnectionType', 'Blockly.utils.Svg', 'Blockly.utils.colour', 'Blockly.utils.dom', 'Blockly.utils.object', 'Blockly.utils.parsing', 'Blockly.utils.svgPaths', 'Blockly.utils.userAgent'], {'lang': 'es6', 'module': 'goog'});
-goog.addDependency('../../core/renderers/common/debug.js', ['Blockly.blockRendering.debug'], [], {'lang': 'es6', 'module': 'goog'});
+goog.addDependency('../../core/renderers/common/debug.js', ['Blockly.blockRendering.debug'], ['Blockly.utils.deprecation'], {'lang': 'es6', 'module': 'goog'});
```
Comment (cpcallen) [inline tests/deps.js:146]: This change does not appear to have anything to do with this PR.  Is it just an update that was inadvertently omitted from a previous PR?  If so fine.

**Review by cpcallen — APPROVED** [review]

File: tests/deps.js (hunk 1 above)
Comment (BeksOmega) [reply tests/deps.js:146]: Looks to be, yesh. Probably the PR that deprecated the debug renderer. Reran build:deps and there were no changes.

**Pull request merged**